Exception ('NoneType') in Tracking workflow

When loading the example project `conservationTracking.ilp` from the [Drosophila 3D+t](http://data.ilastik.org/drosophila.zip) dataset and then clicking the **Track!** button, I get the following error message:

![ilastiktrackingexception-1](https://cloud.githubusercontent.com/assets/2033938/17886575/beb7d414-6922-11e6-9e1e-ab292046504d.png)

This is the stack trace (tested with ilastik-1.2.0rc10 on Windows 7):

```
ERROR 2016-08-23 11:10:49,908 log_exception 5400 6256 Traceback (most recent call last):
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\tracking\conservation\conservationTrackingGui.py", line 263, in _track
    solverName=solver
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\tracking\conservation\opConservationTracking.py", line 220, in track
    if not self.DetectionProbabilities.ready() or len(self.DetectionProbabilities([0]).wait()[0]) == 0:
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 480, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 506, in _wait
    self._wait_within_request( current_request )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 305, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\slot.py", line 850, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\objectClassification\opObjectClassification.py", line 1127, in execute
    tmpfeats = self.Features(times_not_cached).wait()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 480, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 506, in _wait
    self._wait_within_request( current_request )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 305, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\slot.py", line 850, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\objectExtraction\opObjectExtraction.py", line 214, in execute
    val = self.Input(start, stop).wait()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 480, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 506, in _wait
    self._wait_within_request( current_request )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 305, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\slot.py", line 850, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\operators\opArrayCache.py", line 355, in execute
    return self._executeOutput(slot, subindex, roi, result)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\operators\opArrayCache.py", line 469, in _executeOutput
    dirtyPool.wait()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 1181, in wait
    self._clear_finishing_requests()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 1262, in _clear_finishing_requests
    req.block()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 486, in block
    self._wait(timeout) # No return value. Use wait()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 506, in _wait
    self._wait_within_request( current_request )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 305, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 769, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\slot.py", line 850, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\objectExtraction\opObjectExtraction.py", line 585, in execute
    pool.wait()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 1181, in wait
    self._clear_finishing_requests()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 1262, in _clear_finishing_requests
    req.block()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 486, in block
    self._wait(timeout) # No return value. Use wait()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 506, in _wait
    self._wait_within_request( current_request )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 305, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\objectExtraction\opObjectExtraction.py", line 575, in compute_features_for_time_slice
    acc = self._extract(rawVolume, labelVolume)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\objectExtraction\opObjectExtraction.py", line 655, in _extract
    global_features[plugin_name] = plugin.plugin_object.compute_global(image, labels, feature_dict, axes)
AttributeError: 'NoneType' object has no attribute 'plugin_object'
```

Maybe that's related: when going back to the _Division Detection (optional)_ tab, I get the following exception:
```
ERROR 2016-08-23 11:20:16,839 excepthooks 5400 6420 Unhandled exception in thread: 'MainThread'
ERROR 2016-08-23 11:20:16,842 excepthooks 5400 6420 Traceback (most recent call last):
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\shell\gui\ilastikShell.py", line 936, in handleAppletBarItemExpanded
    self.setSelectedAppletDrawer(drawerIndex)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\shell\gui\ilastikShell.py", line 957, in setSelectedAppletDrawer
    self.showCentralWidget(applet_index)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\shell\gui\ilastikShell.py", line 972, in showCentralWidget
    centralWidget = self._applets[applet_index].getMultiLaneGui().centralWidget()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\base\singleToMultiGuiAdapter.py", line 64, in centralWidget
    if self.currentGui() is None:
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\base\singleToMultiGuiAdapter.py", line 44, in currentGui
    self._guis[self._imageLaneIndex] = self.singleImageGuiFactory( self._imageLaneIndex )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\base\standardApplet.py", line 142, in __createSingleLaneGui
    return self.singleLaneGuiClass( self, singleLaneOperator )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\layerViewer\layerViewerGui.py", line 67, in __call__
    instance = super(LayerViewerGuiMetaclass, cls).__call__(*args,**kwargs)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\objectClassification\objectClassificationGui.py", line 174, in __init__
    cfn = self.op.ComputedFeatureNames[:].wait()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 480, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 503, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\request\request.py", line 305, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\lazyflow\lazyflow\slot.py", line 850, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.2.0\ilastik-meta\ilastik\ilastik\applets\trackingFeatureExtraction\opTrackingFeatureExtraction.py", line 218, in execute
    assert plugin_name not in feat_names_div, "feature name dictionaries must be mutually exclusive"
AssertionError: feature name dictionaries must be mutually exclusive
```

--- a/ilastik/applets/base/appletSerializer.py
+++ b/ilastik/applets/base/appletSerializer.py
@@ -787,6 +787,19 @@ class SerialDictSlot(SerialSlot):
         except AssertionError as e:
             warnings.warn('setValue() failed. message: {}'.format(e.message))
 
+class SerialObjectFeatureNamesSlot(SerialDictSlot):
+    """Backwards compatible serializer for DictSlot containing feature names"""
+
+    def _getValue(self, subgroup, slot):
+        """Retrieves value for Slot "slot" from the h5 subgroup "subgroup"
+
+        Global feature names used to be saved into .ilp files under a '0' key.
+        That is no longer the case, so this method peels that extra level off when
+        it is present."""
+        if list(subgroup.keys()) == ['0']:
+            subgroup = subgroup['0']
+        return super()._getValue(subgroup, slot)
+
 class SerialClassifierFactorySlot(SerialSlot):
     def __init__(self, slot, name=None):
         super( SerialClassifierFactorySlot, self ).__init__( slot, name=name )

--- a/ilastik/applets/objectExtraction/objectExtractionSerializer.py
+++ b/ilastik/applets/objectExtraction/objectExtractionSerializer.py
@@ -30,7 +30,8 @@ from lazyflow.roi import getIntersectingBlocks, TinyVector, getBlockBounds, roiT
 from lazyflow.request import Request, RequestLock, RequestPool
 
 from ilastik.applets.base.appletSerializer import AppletSerializer,\
-    deleteIfPresent, getOrCreateGroup, SerialSlot, SerialBlockSlot, SerialDictSlot
+    deleteIfPresent, getOrCreateGroup, SerialSlot, SerialBlockSlot, \
+    SerialDictSlot, SerialObjectFeatureNamesSlot
 from ilastik.utility.commandLineProcessing import convertStringToList
 
 logger = logging.getLogger(__name__)
@@ -109,7 +110,7 @@ class ObjectExtractionSerializer(AppletSerializer):
                             selfdepends=False,
                             shrink_to_bb=False,
                             compression_level=1),
-            SerialDictSlot(operator.Features),
+            SerialObjectFeatureNamesSlot(operator.Features),
             SerialObjectFeaturesSlot(operator.BlockwiseRegionFeatures,
                                      operator.RegionFeaturesCacheInput,
                                      operator.RegionFeaturesCleanBlocks,

--- a/ilastik/applets/trackingFeatureExtraction/trackingFeatureExtractionSerializer.py
+++ b/ilastik/applets/trackingFeatureExtraction/trackingFeatureExtractionSerializer.py
@@ -1,5 +1,5 @@
 from ilastik.applets.base.appletSerializer import AppletSerializer, SerialSlot,\
-    deleteIfPresent, getOrCreateGroup, SerialBlockSlot, SerialDictSlot
+    deleteIfPresent, getOrCreateGroup, SerialBlockSlot, SerialDictSlot, SerialObjectFeatureNamesSlot
 from ilastik.applets.objectExtraction.objectExtractionSerializer import ObjectExtractionSerializer,\
     SerialObjectFeaturesSlot
 
@@ -17,8 +17,8 @@ class TrackingFeatureExtractionSerializer(AppletSerializer):
                             selfdepends=False,
                             shrink_to_bb=False,
                             compression_level=1),
-            SerialDictSlot(operator.FeatureNamesVigra),
-            SerialDictSlot(operator.FeatureNamesDivision),
+            SerialObjectFeatureNamesSlot(operator.FeatureNamesVigra),
+            SerialObjectFeatureNamesSlot(operator.FeatureNamesDivision),
             SerialObjectFeaturesSlot(operator.BlockwiseRegionFeaturesVigra,
                                      operator.RegionFeaturesCacheInputVigra,
                                      operator.RegionFeaturesCleanBlocksVigra,

--- a/tests/test_applets/base/testSerializer.py
+++ b/tests/test_applets/base/testSerializer.py
@@ -25,9 +25,13 @@ import vigra
 import unittest
 import shutil
 import tempfile
+from copy import deepcopy
+from ilastik.applets.base.appletSerializer import SerialObjectFeatureNamesSlot
 from lazyflow.graph import Graph, Operator, InputSlot, Slot, OperatorWrapper
 from lazyflow.operators import OpCompressedUserLabelArray
 from lazyflow.operators.opArrayPiper import OpArrayPiper
+from lazyflow.stype import Opaque
+from lazyflow.rtype import List
 
 from ilastik.applets.base.appletSerializer import \
     getOrCreateGroup, deleteIfPresent, \
@@ -39,6 +43,7 @@ class OpMock(Operator):
     TestSlot = InputSlot(name="TestSlot")
     TestMultiSlot = InputSlot(name="TestMultiSlot", level=1)
     TestListSlot = InputSlot(name='TestListSlot')
+    OpaqueListSlot = InputSlot(rtype=List, stype=Opaque)
 
     def __init__(self, *args, **kwargs):
         super(OpMock, self).__init__(*args, **kwargs)
@@ -329,6 +334,49 @@ class TestSerialDictSlot(unittest.TestCase):
         self.assertTrue( d2_read['a'] == 'A' )
         self.assertTrue( d2_read['b'] == 'B' )
         self.assertTrue( d2_read['c'] == 'C' )
+
+class TestSerialObjectFeatureNamesSlot(unittest.TestCase):
+    def setUp(self):
+        self.mockInput = {
+            "Standard Object Features": {
+                "Coord<Principal<Kurtosis>>": {},
+                "Coord<Principal<Skewness>>": {},
+                "Count": {},
+                "Kurtosis": {},
+                "Maximum": {},
+                "Mean": {},
+                "Minimum": {},
+                "Quantiles": {"something": 123456},
+                "RegionCenter": {},
+                "RegionRadii": {},
+                "Skewness": {},
+                "Sum": {},
+                "Variance": {}
+            }
+        }
+        self.legacyMockInput = {'0': deepcopy(self.mockInput)}
+        self.operator = OpMock(graph=Graph())
+        self.objFeaturesNameSlot = self.operator.OpaqueListSlot
+        self.serializer = SerialObjectFeatureNamesSlot(self.objFeaturesNameSlot)
+        self.testFilePath = os.path.join(tempfile.gettempdir(), 'objFeatureNames.h5')
+
+
+    def testSerialization(self):
+        for inp in (self.mockInput, self.legacyMockInput):
+            h5file = h5py.File(self.testFilePath, 'w')
+            h5group = h5file.create_group('Some Group')
+
+            self.objFeaturesNameSlot.setValue(deepcopy(inp))
+            self.assertEqual(self.objFeaturesNameSlot([]).wait(), inp)
+            self.serializer.serialize(h5group)
+
+            self.objFeaturesNameSlot.setValue({"something": "else"})
+            self.assertNotEqual(self.objFeaturesNameSlot([]).wait(), inp)
+
+            self.serializer.deserialize(h5group)
+            self.assertEqual(self.objFeaturesNameSlot([]).wait(), self.mockInput)
+            h5file.close()
+
 
 class TestSerialBlockSlot(unittest.TestCase):
     

